qualification/Jenkinsfile

```diff
@@ -0,0 +1,147 @@
+       * NOTE: The driver is not always ubverbs0, sometimes it is ubverbs1 or
+       * 2 etc. The correct once needs to be specified.
+       */
+      args '--gpus=all -v /var/run/docker.sock:/var/run/docker.sock --network=host -u=root --ulimit=memlock=-1 --device=/dev/infiniband/rdma_cm  --device=/dev/infiniband/uverbs0'
+    }
+  }
+
+  options {
+    timeout(time: 1, unit: 'HOURS')
+  }
+
+  stages {
+    stage('Install Python packages') {
+      steps {
+        sh 'pip install -r requirements.txt -r requirements-dev.txt'
+      }
+    }
+
+    stage('Install katgpucbf package') {
+      steps {
+        sh 'pip install --no-deps ".[test]" && pip check'
+      }
+    }
+
+    stage('Parallel stage') {
+      parallel {
+        stage('Compile C++ tools') {
+          steps {
+            // Make and compile tools.
+            dir('src/tools') {
+              sh 'make clean'
+              sh 'make -j'
+            }
+          }
+        }
+
+        /* This stage ensures that all the python style guidelines checks pass.
+         * This will catch if someone has committed to the repo without
+         * installing the required pre-commit hooks, or has bypassed them.
+         */
+        stage('Run pre-commit checks') {
+          steps {
+            sh 'pre-commit install'
+            // no-commit-to-branch complains if we are on the main branch
+            sh 'SKIP=no-commit-to-branch pre-commit run --all-files'
+          }
+        }
+      }
+    }
+
+    /* This stage executes the qualification tests on the latest docker image
+     * of katgpucbf that has been previously unit-tested, and generates a report of the
+     * qualification test results.
+     */
+    stage('Run qualification tests') {
+       steps {
+         script {
+           sh 'pip install -r qualification/requirements.txt'
+           sh "spead2_net_raw pytest -v qualification --image-override katgpucbf:harbor.sdp.kat.ac.za/cbf/katgpucbf:latest"
+           sh "qualification/report/generate_pdf.py report.json report.pdf"
+         }
+       }
+    }
+  }
+
+  /* This post stage is configured to always run at the end of this pipeline,
+   * regardless of the completion status. In this stage an email is sent to
+   * the specified address with details of the Jenkins job and a pdf report
+   * containing the qualification test results. The final step removes the workspace when
+   * the build is complete.
+   */
+  post {
+    always {
+      emailext attachLog: true,
+      attachmentsPattern: 'report.pdf',
+      body: """<b>Overall Test Results:</b> ${env.JOB_NAME} - Build#${env.BUILD_NUMBER} - ${currentBuild.result}<br>
+      <b>Node:</b> ${env.NODE_NAME}<br>
+      <b>Duration:</b> ${currentBuild.durationString}<br>
+      <b>Build URL:</b> ${env.BUILD_URL}<br>
+      <br>
+      <i>Note: This is an Automated email notification.</i>""",
+      recipientProviders: [developers(), requestor(), culprits()],
+      subject: '$PROJECT_NAME - $BUILD_STATUS!',
```

Comment:
I'd suggest changing this to explicitly reflect that it's the qualification test.